cli/flox-rust-sdk/src/providers/catalog.rs

```diff
@@ -753,9 +813,14 @@ mod tests {
             NonZeroU32::new(3).unwrap(),
         );
 
-        let collected: Vec<i32> = stream.try_collect().await.unwrap();
+        let collected: Vec<StreamItem<i32>> = stream.try_collect().await.unwrap();
 
-        assert_eq!(collected, (1..=3).collect::<Vec<_>>());
+        assert_eq!(collected, [
+            StreamItem::TotalCount(3),
+            StreamItem::Result(1),
+            StreamItem::Result(2),
+            StreamItem::Result(3)
+        ]);
```

Comment:
🚧 I gave up finding the funcitonal way to do this and the other tests need updating to include TotalCount.